### PR TITLE
Updated tests for sagemaker-code-editor

### PR DESCRIPTION
**Pull request by aws-prayags**
…rectly; Updated tests for sagemaker-code-editor

*Issue #, if available:*

*Description of changes:*
1. Updated Dockerfile so that all the extensions test file is parsed correctly regardless of a new line at the end 
2. Updated tests for sagemaker-code-editor to check for extensions, settings file
3. Updated CONTRIBUTING.md to include common pytest commands

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -90,6 +90,17 @@ Follow these steps for sending out a pull request for adding new packages:
    that we can look more into it.
 7. Add the relevant tests in https://github.com/aws/sagemaker-distribution/blob/main/test/test_dockerfile_based_harness.py 
     and run the build command once again without `--skip-tests` flag.
+   ```
+   When writing or debugging tests, you can use standard pytest commands and arguments (https://docs.pytest.org/en/8.0.x/how-to/usage.html) to run specific tests and change test execution behavior. Some useful commands:
```
Comment (claytonparnell) [inline CONTRIBUTING.md:94]: nit- if this is in code block can you make it a comment?

File: template/v1/Dockerfile [hunk 2]
```diff
@@ -57,7 +57,7 @@ RUN rsync -a ${DIRECTORY_TREE_STAGE_DIR}/ / && \
 
 # CodeEditor - download the extensions
 RUN mkdir -p /etc/code-editor/extensions && \
-    while IFS= read -r url; do \
+    while IFS= read -r url || [ -n "$url" ]; do \
```
Comment (aws-prayags) [inline template/v1/Dockerfile:60]: this change ensures that the last line is also processed (regardless of a new line at end)

**Review by claytonparnell — APPROVED** [review]

**Review by balajisankar15 — APPROVED** [review]

**Comment by aws-prayags** [comment]
Ran the tests with new image `python -m pytest -n auto -m cpu -v -rs -k "sagemaker-code-editor" --local-image-version $VERSION`. 
Tests pass successfully.

File: test/test_artifacts/v1/scripts/run_sagemaker_code_editor_tests.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status.
+set -e
```
Comment (aws-spenceng) [inline test/test_artifacts/v1/scripts/run_sagemaker_code_editor_tests.sh:4]: Do we want to add logging here too with `-x`?

File: test/test_artifacts/v1/scripts/run_sagemaker_code_editor_tests.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+done
+echo "Verified that all extension folders are present in $extensions_base_dir."
+
+# Check that settings file is copied
+MACHINE_SETTINGS_FILE_PATH="/opt/amazon/sagemaker/sagemaker-code-editor-server-data/data/Machine/settings.json"
+if [ ! -f "$MACHINE_SETTINGS_FILE_PATH" ]; then
+    echo "Error: Settings file does not exist at $MACHINE_SETTINGS_FILE_PATH."
+    exit 1
+fi
+
```
Comment (aws-spenceng) [inline test/test_artifacts/v1/scripts/run_sagemaker_code_editor_tests.sh:60]: nit: shouldn't be empty line here at line 60

**Review by aws-spenceng — APPROVED** [review]

**Review by claytonparnell — APPROVED** [review]

**Review by aws-pangestu — APPROVED** [review]

**Pull request merged**